org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/configuration/BuildConfiguration.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2018 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.buildship.core.configuration;
+
+import java.io.File;
+
+import org.eclipse.buildship.core.internal.configuration.DefaultBuildConfigurationBuilder;
+
+/**
+ * Describes a configuration of a Gradle build.
+ *
+ * @author Donat Csikos
+ * @since 3.0
+ */
+public abstract class BuildConfiguration {
```

Comment:
❌ Shouldn't be abstract. Instead give it a private constructor to prevent instantiation.

Reply:
I want to build a proper builder pattern where all the implementation details are hidden in the concrete class. How about adding the `@noimplement` annotation and a protected constructor?

Reply:
This is all unnecessary. The builder pattern does not require an interface. It just requires private constructors.

Reply:
I know how it works, I only wanted add a twist to it :)

Reply:
Fixed.